c error: result of function returning an optional cannot be returned directly

**V version:** V 0.1.27 1e504fb
**OS:** Manjaro Linux

**What did you do?**
```v
fn my_func() string {
  return os.read_file('/some/file/that/exists.txt') or panic(err)
}
```

**What did you expect to see?**
In my opinion this should work. Either it returns the file content or panics.
 
**What did you see instead?**
```
/home/lukas/.cache/v/examples_test.tmp.c:9851:9: error: expected expression before ‘Option_string’
 9851 |  return Option_string _t1 = os__read_file(_STR("%.*s\000/examples/%.*s", 2, os__getwd(), name));
      |         ^~~~~~~~~~~~~
/home/lukas/.cache/v/examples_test.tmp.c:9852:7: error: ‘_t1’ undeclared (first use in this function)
 9852 |  if (!_t1.ok) {
      |       ^~~
/home/lukas/.cache/v/examples_test.tmp.c:9852:7: note: each undeclared identifier is reported only once for each function it appears in
/home/lukas/.cache/v/examples_test.tmp.c: In function ‘_vinit’:
/home/lukas/.cache/v/examples_test.tmp.c:9904:38: warning: integer constant is so large that it is unsigned
 9904 |  _const_math__bits__max_u64 = ((u64)(18446744073709551615));
      |                                      ^~~~~~~~~~~~~~~~~~~~
...
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1258,6 +1258,11 @@ pub fn (mut c Checker) return_stmt(mut return_stmt ast.Return) {
 	}
 	for i, exp_type in expected_types {
 		got_typ := got_types[i]
+		if got_typ.flag_is(.optional) &&
+			(!exp_type.flag_is(.optional) || c.table.type_to_str(got_typ) != c.table.type_to_str(exp_type)) {
+			pos := return_stmt.exprs[i].position()
+			c.error('cannot use `${c.table.type_to_str(got_typ)}` as type `${c.table.type_to_str(exp_type)}` in return argument', pos)
+		}
 		is_generic := exp_type == table.t_type
 		ok := if is_generic { c.check_types(got_typ, c.cur_generic_type) || got_typ == exp_type } else { c.check_types(got_typ,
 				exp_type) }

--- a/vlib/v/checker/tests/return_optional_err.out
+++ b/vlib/v/checker/tests/return_optional_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/return_optional_err.v:4:13: error: cannot use `?string` as type `string` in return argument
+    2 |
+    3 | fn my_func() string {
+    4 |   return os.read_file('/some/file/that/exists.txt') or {panic(err)}
+      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    5 | }
+    6 |

--- a/vlib/v/checker/tests/return_optional_err.vv
+++ b/vlib/v/checker/tests/return_optional_err.vv
@@ -1,0 +1,10 @@
+import os
+
+fn my_func() string {
+  return os.read_file('/some/file/that/exists.txt') or {panic(err)}
+}
+
+fn main() {
+	aa := my_func()
+	println(aa)
+}

